YubiKitTests/Tests/FIDO2Tests.swift

```diff
@@ -17,19 +17,134 @@ import Foundation
 
 class FIDO2Tests: XCTestCase {
     
+    // Resetting the FIDO2 over lightning requires the user to remove and then insert the key again. This would make the FIDO2 test
+    // not that usable so we only test features that requires a reset on the NFC key.
     func testResetOverNFC() {
         runYubiKitTest { connection, completion in
             if connection as? YKFNFCConnection != nil {
-                connection.fido2Session { session, error in
-                    guard let session = session else { XCTAssertTrue(false, "🔴 Failed to get FIDO2 session: \(error!)"); return }
+                connection.fido2TestSession { session in
                     session.reset { error in
                         guard error == nil else { XCTAssertTrue(false, "🔴 Failed to reset FIDO2 session: \(error!)"); return }
                         print("✅ FIDO2 reset")
                         completion()
                     }
                 }
             } else {
-                print("✅ Skipping FIDO2 reset over lightning")
+                print("✅ Skipping FIDO2 testResetOverNFC() over lightning")
+                completion()
+            }
+        }
+    }
+    
+    
+    func testWrongPinOverNFC() {
+        runYubiKitTest { connection, completion in
+            if connection as? YKFNFCConnection != nil {
+                connection.fido2TestSession { session in
+                    session.setPin("123456") { error in
+                        session.verifyPin("234567") { error in
+                            if let error = error {
```

Comment:
I don't understand this - isn't one of the error variables shadowing the previous one?

Reply:
Yes, the second is shadowing the first. We don't care about the first so it's ok in this case. But I agree this is not great form so I changed it to explicitly ignore the first error.

Reply:
great 👍🏻